raiden/network/proxies/token_network.py

```diff
@@ -194,56 +219,113 @@ def _new_netting_channel(self, partner: typing.Address, settle_timeout: int):
-    ) -> Dict:
+            channel_identifier: typing.ChannelID,
+            participant: typing.Address,
+            partner: typing.Address,
+    ) -> ParticipantDetails:
         """ Returns a dictionary with the channel participant information. """
+
         data = self._call_and_check_result(
             'getChannelParticipantInfo',
-            to_checksum_address(participant1),
-            to_checksum_address(participant2),
+            channel_identifier,
+            to_checksum_address(participant),
+            to_checksum_address(partner),
+        )
+        return ParticipantDetails(
+            address=participant,
+            deposit=data[ParticipantInfoIndex.DEPOSIT],
+            withdrawn=data[ParticipantInfoIndex.WITHDRAWN],
+            is_closer=data[ParticipantInfoIndex.IS_CLOSER],
+            balance_hash=data[ParticipantInfoIndex.BALANCE_HASH],
+            nonce=data[ParticipantInfoIndex.NONCE],
+        )
+
+    def detail_channel(
+            self,
+            participant1: typing.Address,
+            participant2: typing.Address,
+            channel_identifier: typing.ChannelID = None,
+    ) -> ChannelData:
+        """ Returns a dictionary with the channel specific information.
+
+        If no specific channel_identifier is given then it tries to see if there
+        is a currently open channel and uses that identifier.
+
+        """
+        channel_identifier = self._inspect_channel_identifier(
+            participant1=participant1,
+            participant2=participant2,
+            called_by_fn='detail_channel',
+            channel_identifier=channel_identifier,
         )
-        return {
-            'deposit': data[0],
-            'withdrawn': data[1],
-            'is_closer': data[2],
-            'balance_hash': data[3],
-            'nonce': data[4],
-        }
 
-    def detail_channel(self, participant1: typing.Address, participant2: typing.Address) -> Dict:
-        """ Returns a dictionary with the channel specific information. """
         channel_data = self._call_and_check_result(
             'getChannelInfo',
+            channel_identifier,
             to_checksum_address(participant1),
             to_checksum_address(participant2),
         )
 
-        assert isinstance(channel_data[0], typing.T_ChannelID)
-
-        return {
-            'channel_identifier': channel_data[0],
-            'settle_block_number': channel_data[1],
-            'state': channel_data[2],
-        }
+        return ChannelData(
+            channel_identifier=channel_identifier,
+            settle_block_number=channel_data[ChannelInfoIndex.SETTLE_BLOCK],
+            state=channel_data[ChannelInfoIndex.STATE],
+        )
 
     def detail_participants(
             self,
             participant1: typing.Address,
             participant2: typing.Address,
-    ) -> Dict:
+            channel_identifier: typing.ChannelID = None,
+    ) -> ParticipantsDetails:
```

Comment:
Returns `ParticipantDetails` instead of "dictionary"

Reply:
??